cmake/scripts/common/HandleDepends.cmake

```diff
@@ -116,8 +116,18 @@ function(add_addon_depends addon searchpath)
           if(NOT PATCH_PROGRAM OR "${PATCH_PROGRAM}" STREQUAL "")
             if(NOT PATCH_EXECUTABLE)
               # find the path to the patch executable
-              find_program(PATCH_EXECUTABLE NAMES patch)
 
+              if(WIN32)
+                # On Windows prioritize Git patch.exe
+                find_package(Git)
+                if(Git_FOUND)
+                  get_filename_component(GIT_DIR ${GIT_EXECUTABLE} DIRECTORY)
+                  get_filename_component(GIT_DIR ${GIT_DIR} DIRECTORY)
```

Comment:
how come this is needed?

I thought just the first would give you the directory without the filename?

Reply:
I not know exactly why but not works removing second `get_filename_component`:

```
-- Processing windowsstore specific tool: patch
-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.35.3.windows.1")
-- patch utility found at C:/Strawberry/c/bin/patch.exe
```

Is found Strawberry patch  😐

umm probably is because patch.exe is no in same path that Git.exe

patch.exe ---> C:\Program Files\Git\usr\bin
Git.exe ---> C:\Program Files\Git\cmd

I'm not CMake expert but probably `get_filename_component(GIT_DIR ${GIT_DIR} DIRECTORY)` is used as `cd ..` here.

Reply:
Updated with:
`find_program(PATCH_FOUND NAMES patch.exe HINTS ${GIT_DIR} PATH_SUFFIXES ../usr/bin)`

Seems it works in all cases and is more clear that is using parent dir.